src/main/java/com/icthh/xm/ms/configuration/repository/impl/DynamicConfigRepository.java

```diff
@@ -0,0 +1,189 @@
+package com.icthh.xm.ms.configuration.repository.impl;
+
+import com.icthh.xm.commons.config.domain.Configuration;
+import com.icthh.xm.ms.configuration.config.ApplicationProperties.S3Rules;
+import com.icthh.xm.ms.configuration.domain.ConfigVersion;
+import com.icthh.xm.ms.configuration.domain.ConfigurationItem;
+import com.icthh.xm.ms.configuration.domain.ConfigurationList;
+import com.icthh.xm.ms.configuration.repository.PersistenceConfigRepository;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DynamicConfigRepository implements PersistenceConfigRepository {
+
+    private static final ConfigVersion S3_VERSION = new ConfigVersion("s3");
+
+    private final PersistenceConfigRepository jGitRepository;
+    private final PersistenceConfigRepository s3Repository;
+    private final S3Rules s3Rules;
+
+    public DynamicConfigRepository(PersistenceConfigRepository jGitRepository, PersistenceConfigRepository s3Repository, S3Rules s3Rules) {
+        this.jGitRepository = jGitRepository;
+        this.s3Repository = s3Repository;
+        this.s3Rules = s3Rules;
+    }
+
+    @Override
+    public boolean hasVersion(ConfigVersion version) {
+        return jGitRepository.hasVersion(version);
+    }
+
+    @Override
+    public ConfigurationList findAll() {
+        var jGitConfigurationList = jGitRepository.findAll();
+        var jGitData = jGitConfigurationList.getData();
+        var s3ConfigurationList = s3Repository.findAll();
+        var s3Data = s3ConfigurationList.getData();
+        var jGitVersion = jGitConfigurationList.getVersion();
+        return getConfigurationList(s3Data, jGitData, jGitVersion);
+    }
+
+    @Override
+    public ConfigurationList findAllInTenant(String tenantKey) {
+        var jGitConfigurationList = jGitRepository.findAllInTenant(tenantKey);
+        var jGitData = jGitConfigurationList.getData();
+        var s3ConfigurationList = s3Repository.findAllInTenant(tenantKey);
+        var s3Data = s3ConfigurationList.getData();
+        var jGitVersion = jGitConfigurationList.getVersion();
+        return getConfigurationList(s3Data, jGitData, jGitVersion);
+    }
+
+    @Override
+    public ConfigurationList findAllInTenants(Set<String> tenants) {
+        var jGitConfigurationList = jGitRepository.findAllInTenants(tenants);
+        var jGitData = jGitConfigurationList.getData();
+        var s3ConfigurationList = s3Repository.findAllInTenants(tenants);
+        var s3Data = s3ConfigurationList.getData();
+        var jGitVersion = jGitConfigurationList.getVersion();
+        return getConfigurationList(s3Data, jGitData, jGitVersion);
+    }
```

Comment:
These methods looks pretty similar (almost duplicates). Can you consider refactoring?